### ISSUE TITLE
"No supported @:from" when there is a from

### ISSUE THREAD

**Issue by Gama11** [elnabo/json2object#24]
Due to some recent changes (not sure which commit exactly), I also can't compile [vshaxe-build](https://github.com/vshaxe/vshaxe-build) anymore:

>C:\HaxeToolkit\haxe\lib\json2object/git/src/json2object/DataBuilder.hx:248: characters 57-85 : json2object: No parser can be generated for vshaxeBuild.project.ArrayHandle<vshaxeBuild.project.Haxelib> as it has no suppo
rted @:from

The [abstract in question](https://github.com/vshaxe/vshaxe-build/blob/9a66fdd08d43d764504a54f50f808591306e98b6/api/vshaxeBuild/project/Project.hx#L96) does indeed not have a `@:from` _function_, but there _is_ a `from Array<T>`, which used to work fine. Is there a reason why this has been disallowed or is this an oversight?

Here's an isolated example:

```haxe
class Main {
	public static function main() {
		new json2object.JsonParser<Project>();
	}
}

typedef Project = {
	var haxelibs:ArrayHandle<Haxelib>;
}
typedef Haxelib = {}
abstract ArrayHandle<T>(Array<T>) from Array<T> {}
```

**Comment by elnabo** [comment]
It's an oversight of the new code which support @:from from a subset of type.

A fix should come soon.

**Comment by elnabo** [comment]
Ok, it should be fixed by 1361889f866c90c479af8e472e7f4ae041463756.

**Comment by Gama11** [comment]
Can confirm. 👍
